test/e2e/network/traffic_policy.go

```diff
@@ -0,0 +1,453 @@
+// need at least 3 nodes, 2 in one zone and 1 in another
+// 2 in a zone to verify Local vs Cluster and sep zones to verify topology hints
+var _ = common.SIGDescribe("Topology Hints Enabled[Feature:Topology Hints][Feature:ServiceInternalTrafficPolicy]", func() {
+	f := framework.NewDefaultFramework("traffic-policy")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	// filled in BeforeEach
+	var c clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		e2eskipper.SkipUnlessMultizone(c)
+		e2eskipper.SkipUnlessNodeCountIsAtLeast(3)
+	})
+
+	ginkgo.It("honors hints when ITP and ETP use Cluster policies", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeCluster, v1.ServiceInternalTrafficPolicyCluster)
+
+		ginkgo.By("using the endpoint zone for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawMoreThanOnePod)
+
+		ginkgo.By("using the endpoint zone for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, tr.endpointZone, tr.endpointNode.Name, sawMoreThanOnePod)
+	})
+
+	ginkgo.It("honors hints only for internal traffic when ETP is local", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeLocal, v1.ServiceInternalTrafficPolicyCluster)
+
+		ginkgo.By("using the endpoint node for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawOnePodOnEndpointNode)
+
+		ginkgo.By("using the endpoint zone for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, tr.endpointZone, tr.endpointNode.Name, sawMoreThanOnePod)
+	})
+
+	ginkgo.It("honors hints only for external requests when ITP is local", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeCluster, v1.ServiceInternalTrafficPolicyLocal)
+
+		ginkgo.By("using the endpoint zone for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawMoreThanOnePod)
+
+		ginkgo.By("using the endpoint node for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, "", tr.endpointNode.Name, sawOnePodOnEndpointNode)
+	})
+
+	ginkgo.It("ignores hints when ITP and ETP are both local", func() {
+		tr := setupAgnhostDaemonset(c, f, true, v1.ServiceExternalTrafficPolicyTypeLocal, v1.ServiceInternalTrafficPolicyLocal)
+
+		ginkgo.By("using the endpoint node for external requests")
+		checkEndpointHost(c, f, tr, tr.nodeportEndpoint, tr.endpointZone, tr.externalNode.Name, sawOnePodOnEndpointNode)
+
+		ginkgo.By("using the endpoint node for internal requests")
+		checkEndpointHost(c, f, tr, tr.clusterIPEndpoint, "", tr.endpointNode.Name, sawOnePodOnEndpointNode)
+	})
+})
+
+var _ = common.SIGDescribe("Topology Hints Disbled[Feature:Topology Hints][Feature:ServiceInternalTrafficPolicy]", func() {
```

Comment:
if is disabled remove `Topology Hints Disbled[Feature:Topology Hints]` , I think that this can be `Services Traffic Policy` per example

Reply:
I can change this, but I thought this label may be best since the nodeport service always uses topology hints annotations. The test is meant to run with topology hints feature flag enabled, etc.

With that context, if you would still prefer the change, do you mean change the label to "Services Traffic Policy[Feature:ServiceInternalTrafficPolicy]"?

Reply:
I have to read it better, seems most of my comments were inaccurate , sorry

Reply:
Can we use the same Describe block? 
```
var _ = common.SIGDescribe("[Feature:Topology Hints]", func() {
```

and differentiate on the `It` statements?

Reply:
I updated this to use one describe, but instead of using `It` I used `Context` statements to wrap the hints enabled / disabled groups of tests. I can switch it to It if you prefer, but I like how it reads using Context.